### DISCUSSION TITLE
nginx access logging - see user and authentication failed in log

### DISCUSSION THREAD

**Discussion by v1net** [dani-garcia/vaultwarden#4907] [Q&A]
iam looking for a solution how i can see the user and if the login fails in the access.log file of the vaultwarden?

**Comment by BlackDex** [comment]
Try out a failed login and check the logs i would say.

**Comment by v1net** [comment]
8.86.107 - - [30/Aug/2024:11:51:48 +0200] "POST /identity/connect/token HTTP/2.0" 400 351 "https://vaultwarden..de/" "Mozilla/5.0 (Linux; Android 14) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/128.0.6613.88 Mobile DuckDuckGo/5 Safari/537.36"
8.86.107 - - [30/Aug/2024:11:51:49 +0200] "POST /identity/accounts/prelogin HTTP/2.0" 200 71 "https://vaultwarden..de/" "Mozilla/5.0 (Linux; Android 14) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/128.0.6613.88 Mobile DuckDuckGo/5 Safari/537.36"
886.107 - - [30/Aug/2024:11:51:50 +0200] "POST /identity/connect/token HTTP/2.0" 400 351 "https://vaultwarden..de/" "Mozilla/5.0 (Linux; Android 14) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/128.0.6613.88 Mobile DuckDuckGo/5 Safari/537.36"
886.107 - - [30/Aug/2024:11:51:51 +0200] "POST /identity/accounts/prelogin HTTP/2.0" 200 71 "https://vaultwarden..de/" "Mozilla/5.0 (Linux; Android 14) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/128.0.6613.88 Mobile DuckDuckGo/5 Safari/537.36"
886.107 - - [30/Aug/2024:11:51:51 +0200] "POST /identity/connect/token HTTP/2.0" 429 288 "https://vaultwarden..de/" "Mozilla/5.0 (Linux; Android 14) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/128.0.6613.88 Mobile DuckDuckGo/5 Safari/537.36"

**Comment by v1net** [comment]
the code 429 288 - says the web server is allready blocking the ip for logins - i allready tryed that but i cant see the login name

**Reply by BlackDex** [reply]
That means you tried it to many times already.
And the output of the log looks like this.
```
[2024-08-30 19:28:55.616][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: ::ffff:127.0.0.1. Username: blaat@foobar.com.
```

Also, if you expect the username to appear in the nginx logs, or any other reverse proxy logs, that will not be the case.
They will only show in the logs of Vaultwarden it self, which you can define by setting the `LOG_FILE` variable, or anything else which stores the stdout/stderr of vaultwarden anywhere.

**Comment by v1net** [comment]
EXTENDED_LOGGING=false
LOG_LEVEL=warn


which log settings should i set in the .env Datei?

**Reply by BlackDex** [reply]
Without the `EXTENDED_LOGGING` it would look like this:
```
Username or password is incorrect. Try again. IP: ::ffff:127.0.0.1. Username: foobar@blaat.com.
```

Since the log is on the `ERROR` level, then `warn` level should work just fine.

**Comment by v1net** [comment]
Yeahhh it works

Thanks BlackDex!!!!! 

i changed in the .env fil the LOG_FILE location to the vauldwarden directory 

EXTENDED_LOGGING=true
LOG_FILE=/var/lib/vaultwarden/data/access.log
LOG_LEVEL=warn


I will donate to an NGO for your help

**Comment by v1net** [comment]
/var/log/nginx/vaultwarden.access.log 

i was allways looking to this log but its a different log
